prompts/java-add-graalvm-native-image-support.prompt.md

```diff
@@ -0,0 +1,456 @@
+   }
+   ```
+
+   Or register entire packages:
+   ```java
+   @RegisterForReflection(classNames = {"com.example.package.*"})
+   ```
+
+2. **Resource Inclusion**: Add to `application.properties`:
+   ```properties
+   quarkus.native.resources.includes=config/*.json,templates/**
+   quarkus.native.additional-build-args=--initialize-at-run-time=com.example.RuntimeClass
+   ```
+
+3. **Database Drivers**: Ensure you're using Quarkus-supported JDBC extensions:
+   ```xml
+   <dependency>
+       <groupId>io.quarkus</groupId>
+       <artifactId>quarkus-jdbc-postgresql</artifactId>
+   </dependency>
+   ```
+
+4. **Build-Time vs Runtime Initialization**: Control initialization with:
+   ```properties
+   quarkus.native.additional-build-args=--initialize-at-build-time=com.example.BuildTimeClass
+   quarkus.native.additional-build-args=--initialize-at-run-time=com.example.RuntimeClass
+   ```
+
+5. **Container Image Build**: Use Quarkus container-image extensions:
+   ```properties
+   quarkus.native.container-build=true
+   quarkus.native.builder-image=mandrel
+   ```
+
+### Micronaut
+- Micronaut has built-in GraalVM support with minimal configuration
+- Use `@ReflectionConfig` and `@Introspected` annotations as needed
+- Micronaut's ahead-of-time compilation reduces reflection requirements
+
+**Common Micronaut Native Image Tips:**
+
+1. **Bean Introspection**: Use `@Introspected` for POJOs to avoid reflection:
+   ```java
+   @Introspected
+   public class MyDto {
+       private String name;
+       private int value;
+       // getters and setters
+   }
+   ```
+
+   Or enable package-wide introspection in `application.yml`:
+   ```yaml
+   micronaut:
+     introspection:
+       packages:
+         - com.example.dto
+   ```
+
+2. **Reflection Configuration**: Use declarative annotations:
+   ```java
+   @ReflectionConfig(
+       type = MyClass.class,
+       accessType = ReflectionConfig.AccessType.ALL_DECLARED_CONSTRUCTORS
+   )
+   public class MyConfiguration {
+   }
+   ```
+
+3. **Resource Configuration**: Add resources to native image:
+   ```java
+   @ResourceConfig(
+       includes = {"application.yml", "logback.xml"}
+   )
+   public class ResourceConfiguration {
+   }
+   ```
+
+4. **Native Image Configuration**: In `build.gradle`:
+   ```groovy
+   graalvmNative {
+       binaries {
+           main {
+               buildArgs.add("--initialize-at-build-time=io.micronaut")
+               buildArgs.add("--initialize-at-run-time=io.netty")
+               buildArgs.add("--report-unsupported-elements-at-runtime")
+           }
+       }
+   }
+   ```
+
+5. **HTTP Client Configuration**: For Micronaut HTTP clients, ensure netty is properly configured:
+   ```yaml
+   micronaut:
+     http:
+       client:
+         read-timeout: 30s
+   netty:
+     default:
+       allocator:
+         max-order: 3
+   ```
+
+## Best Practices
+
+- **Start Simple**: Build with `--no-fallback` to catch all native image issues
```

Comment:
This is unfortunately still needed, but the good news is that [`--no-fallback` will no longer be necessary in the future](https://github.com/oracle/graal/commit/e1560d1e439f4aea77aaad345e29104bdec95adb).